File: src/openfl/events/IOErrorEvent.nim
import ./Event
import ./ErrorEvent

type
  IOErrorEvent* = ref object of ErrorEvent

proc newIOErrorEvent*(typeStr: string, bubbles: bool = true, cancelable: bool = false, text: string = "", id: int = 0): IOErrorEvent =
  let self = IOErrorEvent()
  self.`type` = typeStr
  self.bubbles = bubbles
  self.cancelable = cancelable
  self.text = text
  self.errorID = id
  return self

method clone*(self: IOErrorEvent): Event =
  let event = newIOErrorEvent(self.`type`, self.bubbles, self.cancelable, self.text, self.errorID)
  event.eventPhase = self.eventPhase
  event.target = self.target
  event.currentTarget = self.currentTarget
  return event

method toString*(self: IOErrorEvent): string =
  "[IOErrorEvent type=" & self.`type` & " bubbles=" & $self.bubbles & " cancelable=" & $self.cancelable & " text=" & self.text & " errorID=" & $self.errorID & "]"

const
  IO_ERROR* = "ioError"

File: kfl/src/main/kotlin/openfl/events/IOErrorEvent.kt
package openfl.events

class IOErrorEvent(
    type: String,
    bubbles: Boolean = true,
    cancelable: Boolean = false,
    text: String = "",
    id: Int = 0
) : ErrorEvent(type, bubbles, cancelable, text, id) {

    companion object {
        const val IO_ERROR = "ioError"
    }

    override fun clone(): IOErrorEvent {
        val event = IOErrorEvent(type, bubbles, cancelable, text, errorID)
        event.target = target
        event.currentTarget = currentTarget
        event.eventPhase = eventPhase
        return event
    }

    override fun toString(): String {
        return formatToString("IOErrorEvent", "type", "bubbles", "cancelable", "text", "errorID")
    }
}
